src/merge_sort.py

```diff
@@ -0,0 +1,78 @@
+"""Merge Sort Module."""
+
+# MERGE SORT (MS)
+#
+# CodeFellows 401d5
+# Submission Date:
+#
+# Authors:  Colin Lamont <https://github.com/chamberi>
+#           Ben Shields <https://github.com/iamrobinhood12345>
+#
+# URL:
+
+
+def merge_sort(msl):
+    """Merge sort method."""
+    if len(msl) == 1 or not msl:
+        return msl
+    msl1 = msl[:int(len(msl) / 2)]
+    msl2 = msl[int(len(msl) / 2):]
+
+    msl1 = merge_sort(msl1)
+    msl2 = merge_sort(msl2)
+
+    def _merge(msla, mslb):
+        """Merge compares the two lists and returns a sorted list from lowest to highest value.""" 
+        sorted_list = []
+        while len(msla) and len(mslb):
+            if msla[0] < mslb[0]:
+                low = msla.pop(0)
+            else:
+                low = mslb.pop(0)
+            sorted_list.append(low)
+        if len(msla):
+            sorted_list.extend(msla)
+            return sorted_list
+        else:
+            sorted_list.extend(mslb)
+            return sorted_list
+
+    return _merge(msl1, msl2)
+
+
+def _random_list():
+    """Return a list of random numbers from 0 to 300 of random size less than 300."""
+    import random
+    b = random
+    return b.sample(range(0, 300), 150)
+
+
+a = _random_list()
+r = a[:]
+b = sorted(a)
+w = b[::-1]
+
+
+if __name__ == "__main__":
+    import timeit
+
+    random_merge_sort_timed = timeit.repeat(stmt="merge_sort(r)", setup="from merge_sort import merge_sort, r", number=1000, repeat=3)
+    random_average_merge_sort_timed = float(sum(random_merge_sort_timed) / len(random_merge_sort_timed))
+
+    print("number of runs: " + str(3))
+    print("random merge_sort_timed: " + str(random_merge_sort_timed))
+    print("average: ", str(random_average_merge_sort_timed))
+
+    best_merge_sort_timed = timeit.repeat(stmt="merge_sort(b)", setup="from merge_sort import merge_sort, b", number=1000, repeat=3)
+    best_average_merge_sort_timed = float(sum(best_merge_sort_timed) / len(best_merge_sort_timed))
+
+    print("number of runs: " + str(3))
+    print("best case merge_sort_timed: " + str(best_merge_sort_timed))
+    print("average: ", str(best_average_merge_sort_timed))
+
+    worst_merge_sort_timed = timeit.repeat(stmt="merge_sort(w)", setup="from merge_sort import merge_sort, w", number=1000, repeat=3)
+    worst_average_merge_sort_timed = float(sum(worst_merge_sort_timed) / len(worst_merge_sort_timed))
+
```

Comment:
looking at your output, it is clear that the 'worst case' is not the actual worst case.  Simply reversing a list is not the worst case for this algorithm.

number of runs: 3
random merge_sort_timed: [0.754100672000277, 0.7524420129998362, 0.7507033059996502]
average:  0.7524153303332545
number of runs: 3
best case merge_sort_timed: [0.5421898399999918, 0.5447728300000563, 0.5409313089999159]
average:  0.5426313263333213
number of runs: 3
worst case merge_sort_timed: [0.5831306340000992, 0.6116077660003612, 0.5884709070001009]
average:  0.5944031023335204